test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -2027,6 +2027,45 @@ def test_cgroup_cpuset_memory_spread_page(self):
         self.assertEqual(result['rc'], 0)
         self.assertEqual(result['out'][0], '1')
 
+    def test_cgroup_use_hierarchy(self):
+        """
+        Test that memory.use_hierarchy is enabled by default
+        when PBS cgroups hook is instantiated
+        """
+
+        now = int(time.time())
+
+        # Remove PBS directories from memory subsystem
+        if 'memory' in self.paths and self.paths['memory']:
+            cdir = self.paths['memory']
+            if os.path.isdir(cdir):
+                cpath = os.path.join(cdir, 'pbspro')
+                if not os.path.isdir(cpath):
+                    cpath = os.path.join(cdir, 'pbspro.slice')
```

Comment:
One more thing, something to think about for cgroup tests in general, can the dir names 'pbspro.slice, 'memory' or 'pbspro' ever change? I feel like it might be better to have these be variables in some cgroups specific PTL module so that if ever we need to change these, it can be done without any of the tests breaking. What do you think?

Reply:
Thanks Ravi. it is a good thought. directory name pbspro and pbspro.slice has changed between cgroups v1 and cgroups v2, so far it is covered in the test.
For subsystems like memory, cpuset, function get_paths is taking care of it.
A module PTL is a fine idea and we should think about the use cases for that. The only use case I could think off is running other regression tests with cgroups hook enabled where today I have to do all these cleanups and checks manually. we should think more and take it later I feel.